src/vm/core.zig

```diff
@@ -282,37 +282,56 @@ pub const CairoVM = struct {
         self: *Self,
         instruction: *const instructions.Instruction,
     ) !OperandsResult {
-        // Compute the destination address and get value from the memory.
-        const dst_addr = try self.run_context.computeDstAddr(instruction);
-        const dst = try self.segments.memory.get(dst_addr);
-
-        // Compute the OP 0 address and get value from the memory.
-        const op_0_addr = try self.run_context.computeOp0Addr(instruction);
-        // Here we use `catch null` because we want op_0_op to be optional since it's not always used.
-        // TODO: identify if we need to use try or catch here.
-        const op_0_op = try self.segments.memory.get(op_0_addr);
-
-        // Compute the OP 1 address and get value from the memory.
-        const op_1_addr = try self.run_context.computeOp1Addr(
+
+        var op_res = OperandsResult.default();
+
+        op_res.dst_addr = try self.run_context.computeDstAddr(instruction);
+        op_res.dst = try self.segments.memory.get(op_res.dst_addr);
+        op_res.op_0_addr = try self.run_context.computeOp0Addr(instruction);
+        op_res.op_0 = try self.segments.memory.get(op_res.op_0_addr);
+
+        op_res.op_1_addr = try self.run_context.computeOp1Addr(
             instruction,
-            op_0_op,
+            op_res.op_0,
         );
-        const op_1_op = try self.segments.memory.get(op_1_addr);
 
-        const res = try computeRes(instruction, op_0_op, op_1_op);
+        op_res.op_1 = try self.segments.memory.get(op_res.op_1_addr);
+
+        op_res.res = try computeRes(instruction, op_res.op_0, op_res.op_1);
+        const res_op: ?MaybeRelocatable = op_res.res;
 
         // Deduce the operands if they haven't been successfully retrieved from memory.
-        // TODO: Implement this.
 
-        return .{
-            .dst = dst,
-            .res = res,
-            .op_0 = op_0_op,
-            .op_1 = op_1_op,
-            .dst_addr = dst_addr,
-            .op_0_addr = op_0_addr,
-            .op_1_addr = op_1_addr,
-        };
+        const op_1_ptr = &try self.segments.memory.get(op_res.op_1_addr);
+        const dst_ptr = &try self.segments.memory.get(op_res.dst_addr);
+        const dst_op: ?*const MaybeRelocatable = dst_ptr;
+
+        op_res.op_0 = self.segments.memory.get(op_res.op_0_addr) catch
+            try self.computeOp0Deductions(
+                op_res.op_0_addr,
+                instruction,
+                dst_ptr,
+                op_1_ptr,
+            );
+
+        op_res.op_1 = self.segments.memory.get(op_res.op_1_addr) catch
```

Comment:
Same comment as for `op_0` here due to the lines:

```zig
op_res.op_1_addr = try self.run_context.computeOp1Addr(instruction, op_res.op_0);
op_res.op_1 = try self.segments.memory.get(op_res.op_1_addr);
```
that can give rise to an error if `self.segments.memory.get(op_res.op_1_addr)` gives an error no? So I would perhaps recommend putting `catch` patterns directly in the assignments of `op_res.op_0` and `op_res.op_1` above, no?

Reply:
If you do a `catch` at `op_res.op_1 = try self.segments.memory.get(op_res.op_1_addr);` then you need to do `computeOp1Deductions` and this function needs `res` already computed. That's why I divided the first computation of `op_1` and the second one with the `catch`. 

Or you mean that in the first `op_1` we need to do 

```
op_res.op_1 = self.segments.memory.get(op_res.op_1_addr) catch null;
``` 
to assign `null` to `op_1` from the beginning?

Reply:
Yes that would be correct I think to avoid being stop there due to an error but assigning a null instead